### DISCUSSION TITLE
A table filter gets applied even if not selected

### DISCUSSION THREAD

**Discussion by bvicini** [filamentphp/filament#14115] [Help] [answered]
### Package

Table builder

### Package Version

v3.2.99

### How can we help you?

Hello, in my table I added two filters. the first relates to the database column doc_class_id, the second is unrelated to a specific database column, implemented as custom filter. For some reason that I cannot find out, the second filter applies automatically at page load (the url query string is clean). This is my code. Any suggestions?

->filters([
      \Filament\Tables\Filters\SelectFilter::make('doc_class_id')
      ->label('Tip')
      ->options(\App\Models\DocClass::all()->pluck('name', 'id')),
    
      \Filament\Tables\Filters\Filter::make('dummy')
      ->form([
          \Filament\Forms\Components\Select::make('dummy2')
          ->options(\App\Models\SchoolClass::all()->pluck('slug', 'id')),
      ])
      ->query(fn (Builder $query, array $data): Builder => $query->where('school_class_id',1))
])

the query doesn't use $data by purpose, for he sake of simplicity.

**Comment by bvicini** [comment]
uhmmm I think I just missed ->when($data[dummy2]) Could be this the problem here?

**Comment by e2d2-dev** [comment, marked as answer]
missing the ->when like you mentioned and use ->modifyQueryUsing

**Comment by bvicini** [comment]
Oh thanks for suggestion. it solved my problem

**Comment by e2d2-dev** [comment]
muy bien, thanks for marking as an answer
